### ISSUE TITLE
Add the ability to have armor/weapons auto filled in based on perks chosen.

### ISSUE THREAD

**Issue by Bomjus1** [atomicptr/dauntless-builder-reforged#96]
The request is simple, making the feature itself though, is probably waaaay more complicated or else it would probably already be a thing. Anyway, the most difficult aspect of build making I have in Dauntless is that you need to do an intricate dance to get the perks you want for a build. Sometimes to take away +3 of one perk to add +3 to another means I have to swap 3 armor pieces around. Is it possible for the builder to fill in the armor pieces and cells for you from the perks you have selected? Or is this feature beyond the capabilities/scope of the builder?

Either way, couldn't imagine creating/managing builds without your tool. Thank you!

**Comment by IvanKarpan** [comment]
I was just thinking about the same idea...

Since the number of combinations is finite, technically it is possible to process all of them upfront and then show possible combinations to achieve the targeted perk requirements.

The problem is the number of combination is quite high. So storing this data and quickly scanning it would be real hard. And in case the query parameters are too careless (e.g., all builds that have Bladestorm 1) the subset would be ridiculously large.

Intersecting the results with what cells the user actually owns on the fly is yet another heroic undertaking.

Or so it seems.

This is a little bit of a “too bad I didn’t major in math” kind of moment for me.

**Comment by atomicptr** [comment]
This is probably [sum of subsets](https://en.wikipedia.org/wiki/Subset_sum_problem) which is a fairly complex issue, if we'd only go with +3 cells it might be easier to do as the amount of options shrinks but still kinda meh. 

Tho I see the issue, maybe having something like a "recommended gear pieces" which try to highlight gear pieces with perks that you don't have maxed yet. For instance If I have +3 Overpower it could try to recommend me something like Infernos Razor as a weapon as it has +3 Overpower as well -> Which would lead me to have that perk maxed. This could also be extended to cell slots 🤔. Tho generally I would want to avoid having too heavy calculations on the client side.

**Comment by ghost** [comment]
If the user is forced to select 6*(+6 Perk)s, then the list of possible [weapons, armour pieces] can be drastically reduced.

**Comment by vogzott** [comment]
I second this idea. If you could select from possible variations with the same outcome. Would be like prealgebra solving for [x]

**Comment by Xuhybrid** [comment]
I came here to request the same thing. I want to select required perks and have it suggest gear combinations that would fit. I found an application that did this for Monster Hunter World and it was really useful.

I second that it should only be for +3 cells and gear. Have people select the perk and a dropdown for +3 or +6 because everyone will eventually have the +3 and they can just slot a +2 until they have what they want. Nobody is going to intentionally use a lower cell than they are able.
